airbyte/sources/util.py

```diff
@@ -108,26 +111,50 @@ def get_source(
         metadata = get_connector_metadata(name)
     except exc.AirbyteConnectorNotRegisteredError:
         if not pip_url:
+            ex = exc.AirbyteConnectorNotRegisteredError
+            _log_install_state(None, name, state=EventState.FAILED, exception=Exception(ex))
             # We don't have a pip url or registry entry, so we can't install the connector
             raise
 
-    executor = VenvExecutor(
-        name=name,
-        metadata=metadata,
-        target_version=version,
-        pip_url=pip_url,
-    )
-    if install_if_missing:
-        executor.ensure_installation()
+    try:
+        executor = VenvExecutor(
+            name=name,
+            metadata=metadata,
+            target_version=version,
+            pip_url=pip_url,
+        )
+        if install_if_missing:
+            executor.ensure_installation()
 
-    return Source(
-        name=name,
-        config=config,
-        streams=streams,
-        executor=executor,
-    )
+        return Source(
+            name=name,
+            config=config,
+            streams=streams,
+            executor=executor,
+        )
+    except Exception as e:
+        _log_install_state(None, name, state=EventState.FAILED, exception=e)
+        raise
 
 
 __all__ = [
     "get_source",
 ]
+
+
+def _log_install_state(
```

Comment:
This is not a class, but prefixed the method with underscore since it's for internal use.